### ISSUE TITLE
AttributeError: 'S3VFSFile' object has no attribute 'xSectorSize'

### ISSUE THREAD

**Issue by blueshed** [uktrade/sqlite-s3vfs#24]
I found this error on OSX with python3.10 and python3.12

Traceback (most recent call last):
  File "src/vfs.c", line 2580, in apswvfsfile_xSectorSize
AttributeError: 'S3VFSFile' object has no attribute 'xSectorSize'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "sample.py", line 16, in <module>
    with apsw.Connection(key_prefix, vfs=s3vfs.name) as db:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "src/vfs.c", line 2580, in apswvfsfile_xSectorSize
AttributeError: 'S3VFSFile' object has no attribute 'xSectorSize'

I found this error when I ran the sample code in your readme. I was able to workaround it by adding:

```python
class S3VFSFile:
...
    def xSectorSize(self):
            return self._block_size
...
```

to 'sqlite_s3vfs.py'

**Comment by michalc** [comment]
Hi @blueshed,

Thanks for the report, and I can reproduce the error.

I suspect it's due to a change in apsw (a dependency of sqlite-s3vfs). From some brief testing the latest version that seems to work is apsw==3.43.1.0, but anything after gives the error you mention.

Will see if we can fix it (e.g. by incorporating the change you suggest). In the meantime, I recommend using sqlite-s3vfs with apsw 3.43.1.0 (or earlier)

Michal

**Comment by michalc** [comment]
Fixed and releasing now in v0.0.36

**Comment by blueshed** [comment]
So quick - just ran on python3.12 - thank you so much.
